clarifying the value of the 3 BXML verb approaches to interworking (#273)
Bandwidth/api-docs: site/docs/webrtc/voice-iw.md

## Connecting WebRTC to Programmable Voice

> Bandwidth has a demonstrated history of excellence in IP voice technologies, including comprehensive CPaaS Programmable Voice (PV) and Messaging capabilities. The introduction of WebRTC provides the opportunity to combine the strength and diversity of our Programmable Voice (PV) offering with our highly controllable WebRTC to yield a rich and flexible platform that allows our customers to extend their business with a broad range of browser-based communication capabilities.

Reviewer: The second sentence is a little wordy IMO. How about:

Combining WebRTC with our Programmable Voice (PV) offering yields a rich and flexible platform that allows our customers to extend their business with a broad range of browser-based communication capabilities.

One of the steps we have taken to enable this capability suite is to ensure that our PV and WebRTC APIs combine their respective strengths when integrated.

> When PV calls interwork with WebRTC communications, the behavior of the voice call can be augmented by the extensive suite of PV capabilities. This enables your application to create voice calls, receive voice calls, insert services like recording, and control and monitor changes made to these calls. This is done with the use of "BXML" verbs supported by the PV platform.

Reviewer: Did you mean to say:

When PV calls interwork with WebRTC communications, the behavior of the **_WebRTC_** call can be augmented by the extensive suite of PV capabilities

Perhaps change interwork to interconnect or integrate:

When PV calls integrate with WebRTC communications, the behavior of the **_WebRTC_** call can be augmented by the extensive suite of PV capabilities

Reviewer: Nit: This can be accomplished with the use of "BXML" verbs supported by the PV platform

Perhaps hyperlink BXML and/or verbs to the voice docs

Author: 1) The intent was that the resulting end-end call would receive the services, rather than the voice or the WebRTC side, so I replaced the word 'voice' with 'resulting' rather than asserting that the WebRTC call itself received the benefit.  I think that is clearer.
2) I will attempt to link the first instance of Programmable Voice to the related documentation instead.

By combining our WebRTC and programmable voice APIs, you can create composite call flows that meet your needs.

> To create a Programmable Voice call that involves WebRTC endpoints, your appliction establishes a connection from the Programmable Voice network to the WebRTC service.

Reviewer: mispelling in application. Couple minor suggestions:

To create a Programmable Voice call that involves WebRTC endpoints, your application **_first_** establishes a connection from the Programmable Voice network to the WebRTC **_platform_**.

![Summary of interconnection](../../static/img/webrtc-pv-summary.png "summary")

To interconnect a WebRTC participant to the PV network capabilities, your application first creates a WebRTC Participant to host the interconnection. The Participant security token is handed to the PV platform which sends a pre-authenticated SIP INVITE to the WebRTC service. This INVITE includes a UUI header containing the Participant authentication token which enables the system to securely identify and interconnect the PV voice capabilities and the WebRTC participants.

Once an answer is received on this SIP transaction, an interconnection is established between WebRTC and Voice.

> You are able to select differnt overall call behaviors by choosing the BXML verbs that are used in setting up the interconnection. The three call control verbs that can be used are: Transfer, Bridge, and Conference, each of which creates a different voice application experience

Reviewer 2: ```suggestion
You are able to select different overall call behaviors by choosing the BXML verbs that are used in setting up the interconnection. The three call control verbs that can be used are: Transfer, Bridge, and Conference, each of which creates a different voice application experience.
```

    });
```

> There’s a limitation to what can be done with Transfer: the fact that only a single call recognized by the PV platform means that changes to that call will impact the WebRTC interconnection. Once the transfer is complete, you can redirect the call to another destination using the [TransferComplete](https://dev.bandwidth.com/voice/bxml/callbacks/transferComplete.html) callback, however this will result in the removal of the WebRTC leg of the call. To do this, you need to specify the transferCompleteUrl attribute. The above example can be found in [Hello-world repository](https://github.com/Bandwidth-Samples/webrtc-hello-world-ts).

Reviewer 2: ```suggestion
There’s a limitation to what can be done with Transfer: the fact that only a single call recognized by the PV platform means that changes to that call will impact the WebRTC interconnection. Once the transfer is complete, you can redirect the call to another destination using the [TransferComplete](../voice/webhooks/transferComplete) callback, however this will result in the removal of the WebRTC leg of the call. To do this, you need to specify the transferCompleteUrl attribute. The above example can be found in [Hello-world repository](https://github.com/Bandwidth-Samples/webrtc-hello-world-ts).
```

Reviewer 2: The `TransferComplete` link currently links to the old site.

Author: Updated, but let me confirm with the build before you attempt approval.

Reviewer 2: I believe you updated to the wrong link in your change (BXML Transfer vs the TransferComplete webhook). When a suggestion is made in GitHub you can hit the "Commit suggestion" button to add it automatically.

Author: You were too quick - I had not yet completed the submission and checked the build.  Sorry 'bout that - it is good now on the most recent build.

Author: after discussion on slack - removed the link as superfluous

            uui: `${participant.token};encoding=jwt`,
```

This [/bridgeCallAnswered](<https://github.com/Bandwidth-Samples/webrtc-voicebridge-ts/blob/main/src/server.ts#:~:text=*/-,app.post(%22/bridgeCallAnswered%22%2C%20async%20(req%2C%20res)%20%3D%3E%20%7B,%7D)%3B,-/**>) API call links your WebRTC session with Programmable Voice. Then, Bandwidth's Voice API will hit the [/callAnswered](<https://github.com/Bandwidth-Samples/webrtc-voicebridge-ts/blob/main/src/server.ts#:~:text=app.post(%22/callAnswered,%7D)%3B>) endpoint when the separate outgoing call is answered. The two independent calls can be established synchronously or in parallel, providing the flexibility in management of the endpoints. For example, a single WebRTC-side call can be established that connects to a series of PV-side calls, without the need to set up the entire end-end experience for each new voice caller. You can find the complete code and instructions for the sample above in the [bridge repository](https://github.com/Bandwidth-Samples/webrtc-voicebridge-ts).

> If reducing delays in interconnection of voice and WebRTC legs is important, the best option would be to use the bridge verb, which allows setting up the legs in parallel, or establishing one leg early in anticipation of the end to end call. Also, use of the Bridge verb is recommended if PV capabilties and services are needed on the voice phone side of the interconnection.

Reviewer 2: ```suggestion
If reducing delays in interconnection of voice and WebRTC legs is important, the best option would be to use the bridge verb, which allows setting up the legs in parallel, or establishing one leg early in anticipation of the end to end call. Also, use of the Bridge verb is recommended if PV capabilities and services are needed on the voice phone side of the interconnection.
```

## Conference

Although use of the Conference verb has higher costs than the previous methods, it would be the most sustainable option for dealing with a large group of voice callers. If you expect more than two voice callers in one session, the Conference method is effective if Programmable Voice services are needed for those callers. You can have 20 calls in a conference and keep the session for 24 hours. You also have the freedom to add and remove participants from the conference.

![Conference Verb Model](../../static/img/webrtc-pv-conference.png "Conference")

> Conferencing is actively supported by both the WebRTC and Voice platforms, however, if both platforms are interconnected, and the power of PV is needed on the voice-only call legs, use of the Conference BXML verb is valuable. If fine-grained control over media in a conference scenario is needed, or of video is required in the conference, it is likely wise to use the conferencing capability inherent in the WebRTC platform and APIs.

Reviewer 2: ```suggestion
Conferencing is actively supported by both the WebRTC and Voice platforms, however, if both platforms are interconnected, and the power of PV is needed on the voice-only call legs, use of the Conference BXML verb is valuable. If fine-grained control over media in a conference scenario is needed, or if video is required in the conference, it is likely wise to use the conferencing capability inherent in the WebRTC platform and APIs.
```
